exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/parser/CompoundIdentifierConverter.java

```diff
@@ -107,7 +117,7 @@ public SqlNode visitChild(
       }
       SqlNode newOperand = operand.accept(CompoundIdentifierConverter.this);
       enableComplex = localEnableComplex;
-      if (newOperand != operand) {
+      if (! newOperand.equalsDeep(operand, false)) {
```

Comment:
I'm a little confused by this change. Operands are expected to be immutable. As such, identity comparison should be sufficient. Are you trying to avoid excessive rewrites?